### PR TITLE
Environment revamp 1

### LINKED ISSUE #128: Pictoralist 2 infrastructure improvements
Some improvements requested by the software team for the infrastructure of pictoralist and its surroundings:

Main/Pictoralist changes (env var related):

- [ ] Debug flag as env var -> interface with logging module (#8) 
    - [ ] Implement logging alongside FastAPI app (likely a can of worms here)
    - [ ] Add debug, info, and warning/critical logging statements where appropriate
    - [x] Implement info level spec by env var

- [x] Display config specifications by env var
    - [x] Timeframe for display(s)
    - [x] Goal line plotting

Pictoralist caching:

- [x] Error handle about cache folder for testers, should check if extant and create if not
- [x] Stop cluttering/bloat of cache, keep images to 20 or less in folder at a time

### PR DESCRIPTION
**Pull request by mackgalante**
Changes to how configurations are set for PFP instances, now with settings.py and separated .env files for local and remote knowledge loading on startup.

Resolves #128 and #139

### CODE REVIEW

**Comment by pboisver** [comment]
On the use of environment variables (some of this may or may not apply to the current implementation):

* Production configuration by default. E.g. LOGLEVEL=INFO, PICTORALESS=FALSE, SECURE=TRUE, etc. (But see point about avoid defaults for data). Mostly this is because when running in production if we explicitly set something we either want it to be correct or to fail immediately. If I do a fresh checkout and run it on my machine it should run exactly the same as on GCP.
* Avoid defaults for data config. The knowledge base pointed to affects the functional behavior. The principle of least surprise says that if I forget to set the KB location, it should not silently use the production KB. That KB root location should be explicitly set, and the PFP should fail immediately if it is not set. The framework you are using includes a sensible exception thrown when unset non-defaulted config is attempted.
* Dev-Prod parity. Exactly the same code should be run in dev and prod, Environment variables should configure functionality of the same code. Using an IS_PROD boolean that loads a different config file means that you can have a test, customer_a, or next_gen, environment without jumping through some hoops and proliferating .env files. Rarely, but sometimes, we want to distinguish environments with a global environment variable; the use cases are quite limited though. And remember that if I do a fresh checkout and run it on my machine it should run exactly the same as on GCP.
* UX matters in API design including config. If I can set KB_ROOT=file:///foo/bar/kb or KB_ROOT=http://github.org/DisplayLab/knowedge-base and have the rest of the kb related env locations derived, that’s enough. Then I don’t have to say IS_PROD=true|false, I can just treat every environment the same (dev-prod parity). If the environment model is good, I’ll also be able to override a specific location by setting KP_MPM for example.
* Be direct. If the difference between IS_PROD true and false is which .env is loaded the an environment variable called DOT_ENV_FILE with a path can be easier to use. The framework allows that location to be set but I not sure right now how I would point to one of three .env files in my home directory using IS_PROD. And it is not so much that we must support multiple environments in multiple locations, it’s just that the common pattern of using the environment to set paths works for all the use cases.

We can do some more testing and review it. I’m sure I haven’t well enough understood the code yet, but these are some of the things we can discuss when we are testing.

File: settings.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+envpath = os.environ.get('ENV_PATH', default='.env.remote')
+
+# Merge env vars from selected .env to the current environment (respecting unix way):
+os.environ.update(dotenv_values(env_path))
+
+## Settings class to store decoupled instance and build settings:
+class Settings:
+    def __init__(self):
+        # Knowledge settings
+        self.templates = config('templates', cast=str)
```
Comment (mackgalante) [inline settings.py:17]: Removed defaults, should fail if .env.remote is not being reached to respect principle of least surprise

File: settings.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+from dotenv import dotenv_values
+from decouple import config
+import pprint
+import os
+
+## Control switch for production and dev configurations
+# User specifies custom env var set to use on startup, else uses prod environment file
+envpath = os.environ.get('ENV_PATH', default='.env.remote')
```
[the remark this answers is not in the document]
Comment (mackgalante) [reply settings.py:8]: I know this violates bullet 2, but it accomplishes bullet 1 wherein a fresh checkout with zero other command line arguments runs as GCP should.

**Comment by mackgalante** [comment]
Tested changes locally, smoke test passing on my machine :)

**Review by pboisver — APPROVED** [review]

**Pull request merged**